org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/RefactorProcessor.java

```diff
@@ -334,15 +398,27 @@ private boolean getInlineProposal(IInvocationContext context, ASTNode node, Coll
 					// Inline Constant (static final field)
 					if (RefactoringAvailabilityTesterCore.isInlineConstantAvailable((IField) varBinding.getJavaElement())) {
 						InlineConstantRefactoring refactoring = new InlineConstantRefactoring(context.getCompilationUnit(), context.getASTRoot(), context.getSelectionOffset(), context.getSelectionLength());
-						if (refactoring != null && refactoring.checkInitialConditions(new NullProgressMonitor()).isOK() && refactoring.getReferences(new NullProgressMonitor(), new RefactoringStatus()).length > 0) {
```

Comment:
Nice, so with this, we correctly defer these calculations to resolve right ? I noticed `ChangeCorrectionProposalCore.getChange()` only gets called from `codeAction/resolve`. That would definitely improve things both for `javac` and `ecj`.

Reply:
I guess the issue is that `checkInitialConditions` is what determines whether there are any references to inline, but by defering it to `codeAction/resolve`, you end up showing the dialog in cases where it would have been hidden.

Reply:
> I guess the issue is that checkInitialConditions is what determines whether there are any references to inline,

Right.

>  but by defering it to codeAction/resolve, you end up showing the dialog in cases where it would have been hidden. It seems like a small price to pay for improving performance though, and it's not horribly wrong.

We can solve it in a separate issue. The dialog requires a client update.